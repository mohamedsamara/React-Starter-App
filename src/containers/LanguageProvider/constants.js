/*
 *
 * LanguageProvider constants
 *
 */

export const SET_LOCALE = 'src/LanguageProvider/SET_LOCALE';
