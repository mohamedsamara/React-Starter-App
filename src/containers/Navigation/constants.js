/*
 *
 * Navigation constants
 *
 */

export const TOGGLE_DRAWER = 'src/Navigation/TOGGLE_DRAWER';
