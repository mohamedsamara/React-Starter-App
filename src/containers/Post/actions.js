/*
 *
 * Post actions
 *
 */

import {
  FETCH_POSTS_REQUEST,
  FETCH_POSTS_SUCCESS,
  HANLDE_POSTS_FAILURE,
  DELETE_POST_SUCCESS,
} from './constants';

export function fetchPostsRequest(isFetching) {
  return {
    type: FETCH_POSTS_REQUEST,
    isFetching,
  };
}

export function fetchPostsSuccess(posts) {
  return {
    type: FETCH_POSTS_SUCCESS,
    posts,
  };
}

export function handlePostFailure(error) {
  return {
    type: HANLDE_POSTS_FAILURE,
    error,
  };
}

export function deletePostSuccess(index) {
  return {
    type: DELETE_POST_SUCCESS,
    index,
  };
}
