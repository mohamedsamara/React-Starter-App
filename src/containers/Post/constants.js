/*
 *
 * Post constants
 *
 */

export const FETCH_POSTS_REQUEST = 'src/Post/FETCH_POSTS_REQUEST';
export const FETCH_POSTS_SUCCESS = 'src/Post/FETCH_POSTS_SUCCESS';
export const HANLDE_POSTS_FAILURE = 'src/Post/HANLDE_POSTS_FAILURE';
export const DELETE_POST_SUCCESS = 'src/Post/DELETE_POST_SUCCESS';
