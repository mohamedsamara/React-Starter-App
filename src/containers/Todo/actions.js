/*
 *
 * Todo actions
 *
 */

import {
  FETCH_TODOS_REQUEST,
  FETCH_TODOS_SUCCESS,
  FETCH_TODOS_FAILURE
} from './constants';

export function fetchTodosRequest(isFetching) {
  return {
    type: FETCH_TODOS_REQUEST,
    isFetching
  };
}

export function fetchTodosSuccess(todos) {
  return {
    type: FETCH_TODOS_SUCCESS,
    todos
  };
}

export function fetchTodosFailure(error) {
  return {
    type: FETCH_TODOS_FAILURE,
    error
  };
}
