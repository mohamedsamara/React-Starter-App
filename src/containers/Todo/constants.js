/*
 *
 * Todo constants
 *
 */

export const FETCH_TODOS_REQUEST = 'src/Dashboard/FETCH_TODOS_REQUEST';
export const FETCH_TODOS_SUCCESS = 'src/Dashboard/FETCH_TODOS_SUCCESS';
export const FETCH_TODOS_FAILURE = 'src/Dashboard/FETCH_TODOS_FAILURE';
