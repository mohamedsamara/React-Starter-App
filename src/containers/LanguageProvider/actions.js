/*
 *
 * LanguageProvider actions
 *
 */

import { SET_LOCALE } from './constants';

export function handleSetLocale(language) {
  return {
    type: SET_LOCALE,
    locale: language,
  };
}
