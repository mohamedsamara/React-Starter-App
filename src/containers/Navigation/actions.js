/*
 *
 * Navigation actions
 *
 */

import { TOGGLE_DRAWER } from './constants';

export function handleToggleDrawer(isDrawerOpen) {
  return {
    type: TOGGLE_DRAWER,
    isDrawerOpen
  };
}
