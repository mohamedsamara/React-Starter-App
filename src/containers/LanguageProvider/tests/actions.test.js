import { handleSetLocale } from '../actions';
import { SET_LOCALE } from '../constants';

describe('LanguageProvider Actions', () => {
  it('has a type of SET_LOCALE', () => {
    const locale = 'en';

    const expected = {
      type: SET_LOCALE,
      locale,
    };
    expect(handleSetLocale(locale)).toEqual(expected);
    expect(handleSetLocale(locale)).toMatchSnapshot();
  });
});
