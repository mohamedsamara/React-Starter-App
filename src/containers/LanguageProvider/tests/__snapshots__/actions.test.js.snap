// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageProvider Actions has a type of SET_LOCALE 1`] = `
Object {
  "locale": "en",
  "type": "src/LanguageProvider/SET_LOCALE",
}
`;
